auth/keto/policies.go

```diff
@@ -44,8 +46,19 @@ func (c policyAgent) CheckPolicy(ctx context.Context, pr auth.PolicyReq) error {
 	return nil
 }
 
-func (c policyAgent) AddPolicy(ctx context.Context, pr auth.PolicyReq) error {
-	trt := c.writer.TransactRelationTuples
+func (pa policyAgent) AddPolicy(ctx context.Context, pr auth.PolicyReq) error {
+	var ss *acl.Subject
+	switch isSubjectSet(pr.Subject) {
+	case true:
+		namespace, object, relation := parseSubjectSet(pr.Subject)
```

Comment:
Looks like you are always checking if something is subject set before calling `parseSubjectSet` so there is probably no need to call `isSubjectSet` inside `parseSubjectSet`.

Reply:
I removed `isSubjectSet` inside `parseSubjectSet`